packages/mongodb-rag-core/src/contentStore/updateEmbeddedContent.ts

```diff
@@ -18,9 +18,26 @@ export interface EmbedConcurrencyOptions {
   createChunks?: number;
 }
 
+const getHashForFunc = (
+  chunkAlgoHashes: Map<string, string>,
+  f: ChunkFunc,
+  o?: Partial<ChunkOptions>
+): string => {
+  const data = JSON.stringify(o ?? {}) + f.toString();
+  const existingHash = chunkAlgoHashes.get(data);
+  if (existingHash) {
+    return existingHash;
+  }
+  const hash = createHash("sha256");
+  hash.update(data);
+  const digest = hash.digest("hex");
+  chunkAlgoHashes.set(data, digest);
+  return digest;
+};
+
 /**
-  (Re-)embeddedContent the pages in the page store that have changed since the given date
-  and stores the embeddedContent in the embeddedContent store.
+  (Re-)embeddedContent the pages in the page store that have changed (copy change, or chunking algothrim change) 
+  since the given date and stores the embeddedContent in the embeddedContent store.
```

Comment:
"since the given date" is only true if there was a content change, right? IIUC a change to our chunk function/options will ignore the since date?

Might need a reword to reflect that.